app/models/merchant.rb

```diff
@@ -31,4 +31,11 @@ def self.revenue(merchant_id)
     .where(id: merchant_id)
     .sum('item_invoices.quantity * item_invoices.unit_price')
   end
+
+  def self.revenue_date_range(start_date, end_date)
```

Comment:
really excited to have figured this out, this nested joins ensures that i don't receive duplicate rows when doing multiple joins